src/validation.cpp

```diff
@@ -3393,7 +3393,7 @@ bool CChainState::AcceptBlockHeader(const CBlockHeader& block, CValidationState&
             }
         }
     }
-    if (pindex == nullptr)
+    if (pindex == nullptr && !state.IsInvalid())
```

Comment:
Can you explain why you think this change is necessary? It looks to me that all the code paths that lead to an invalid state already return early.